flake.nix

```diff
@@ -49,26 +45,42 @@
         let
           pkgs = pkgsFor system;
 
-          rust = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain;
+          rust = fenix.packages.${system}.fromToolchainFile {
+            file = ./rust-toolchain;
+            sha256 = "sha256-6PfBjfCI9DaNRyGigEmuUP2pcamWsWGc4g7SNEHqD2c=";
+          };
 
-          naersk-lib = naersk.lib."${system}".override {
+          buildRustPackage = (pkgs.makeRustPlatform {
             cargo = rust;
             rustc = rust;
-          };
+          }).buildRustPackage;
         in
         rec {
           # `nix build`
-          packages.${name} = naersk-lib.buildPackage {
+          packages.${name} = buildRustPackage rec {
+            inherit name;
             pname = name;
-            root = ./.;
+            src = nixpkgs.lib.cleanSource ./.;
+            cargoLock.lockFile = ./Cargo.lock;
 
+            preBuildPhases = [ "codeStyleConformanceCheck" ];
+
+            codeStyleConformanceCheck = ''
+              export PATH="${rust}/bin:$PATH";
+              # rustfmt
+              cargo fmt -- --check
+              # clippy - needs to rebuild anyhow, so can as well check before the build
+              cargo clippy --all --all-features --tests -- -D clippy::pedantic -D warnings
+            '';
+
+            # buildDeps
             nativeBuildInputs = with pkgs; [
               pkg-config
               installShellFiles
+              nixFlakes
```

Comment:
this is principly necesarry since `strictDeps = true`. `naersk` simply seems to have been a bit more lenient